src/rdrand.rs

```diff
@@ -31,13 +31,14 @@ unsafe fn rdrand() -> Result<[u8; WORD_SIZE], Error> {
     Err(Error::UNKNOWN)
 }
 
+#[cfg(and(target_env = "sgx", not(target_feature = "rdrand")))]
+compile_error!("SGX targets must enable RDRAND to get randomness");
```

Comment:
What does "enable RDRAND" mean? We know, but I expect it could confuse some users. Better to say "... must enable the `rdrand` feature of `getrandom` crate"?

Reply:
No, it should be "enable `rdrand` target feature for your build" or something similar.

Reply:
Fixed to be `Target feature +rdrnd must be enabled for SGX targets.` Note that `+rdrand` is not a typo, see: https://github.com/rust-lang/rust/blob/0dc9e9c10ca6dc78cba8b9f9b15038c977b10a77/src/librustc_target/spec/x86_64_fortanix_unknown_sgx.rs#L56

Reply:
I think you should use `rdrand`, `rdrnd` is a LLVM name, while `rdrand` is a "vendor" name. And it was decided for Rust compiler to follow "vendor" feature names. IIRC Rust compiler translates `rdrand` to `rdrnd` before moving to LLVM, so compiler users should use the former.

Reply:
The problem is that the target feature provided to rustc *must* be `+rdrnd`. If you provide `+rdrand` it does not work. See [this example](https://rust.godbolt.org/z/TXrbCy), with `+rdrand` the checks are not removed, but with `+rdrnd` they are.

Reply:
Okay, using a target feature for configuration makes sense if does what we need, but why then are you using `#[target_feature(enable = "rdrand")]` in the code (not `rdrnd`)?

Reply:
Because when doing detection in the code you have to use "rdrand" (I know, it's inconsistent). If you use `rdrnd` in the rust code, you get:

```
error: the feature named `rdrnd` is not valid for this target
```

Reply:
I believe it's a `rustc` bug. I've created an issue: rust-lang/rust#61895

Reply:
For the time being we could write something like this:
>`rdrand` target feature must be enabled for SGX targets (e.g. by using `-C target-feature=+rdrnd`, see: rust-lang/rust#61895)

And we will change this message depending on how the issue will be handled .

Reply:
I added something like the above in and added a comment explaining the `rdrand` vs `+rdrnd` disparity.

Reply:
You have a typo in the comment:
>requires "+rdrand" flag

Reply:
Fixed, I really hope https://github.com/rust-lang/rust/issues/49653 gets resolved